File: resources/validateAllLocations.js
require('dotenv').config()

const AGENDA_UID = process.env.AGENDA_UID

const { OaSdk } = require('@openagenda/sdk-js') // OpenAgenda Sdk

const publicKey = process.env.OA_PUBLIC_KEY;
const secretKey = process.env.OA_SECRET_KEY;

const oa = new OaSdk({
  publicKey,
  secretKey,
});


/**
 * Validates all locations (status =1 ) in the OpenAgenda instance that are in the "to be validated" state.
 *
 * @return {Promise<void>} Resolves when all locations have been validated.
 */
(async () => {
  await oa.connect();
  const locations = await oa.locations.list(AGENDA_UID, { state: 0 });
  // console.log("locations - ", locations);
  if (locations && locations.locations && locations.locations.length > 1) {
    for (const location of locations.locations) {
      if (location.state === 0) {
        await oa.locations.patch(AGENDA_UID, location.uid, { state: 1 });
        console.log("Validated location: ", location.name)
      }
    }
    console.log("Total validated locations: ", locations.locations.length)
  } else {
    console.log("No locations require validation.")
  }
})();

File: resources/python/validateAllLocations.py

import pprint
import sys,os
from git import Repo

# Ajoute le dossier "ressources" au sys.path
git_root = Repo(search_parent_directories=True).working_tree_dir
sys.path.insert(0,   os.path.abspath(  os.path.join(  git_root,'resources/python' ) ) )

from HttpRequests import (  retrieve_access_token,
                            get_locations,
                            patch_location)

AGENDA_UID = os.getenv('AGENDA_UID')
BASE_URL = "https://api.openagenda.com/v2/"
OA_SECRET_KEY = os.getenv('OA_SECRET_KEY')

def validate_locations():
    accessToken = retrieve_access_token(OA_SECRET_KEY)
    allLocations= get_locations(accessToken)

    if allLocations and len(allLocations) > 0:
        validated_count = 0
        for location in allLocations :
            if location["state"] == 0:
                try:
                    response=patch_location(accessToken, str(location["uid"]), { 'state': 1 })
                    print("Validated location:", location["name"])
                    validated_count += 1
                except Exception as e:
                    print(f"Error validating location: {location["name"]} - {location['uid']}")
                    print(e)
        print("Total validated locations:", validated_count)
        

if __name__ == "__main__":
    validate_locations()